extmod/modtls_mbedtls.c

```diff
@@ -669,6 +685,51 @@ static mp_obj_t ssl_socket_make_new(mp_obj_ssl_context_t *ssl_context, mp_obj_t
     mbedtls_raise_error(ret);
 }
 
+#if defined(MBEDTLS_SSL_DTLS_HELLO_VERIFY) && defined(MBEDTLS_SSL_SRV_C)
+// DTLS Server session cookie support
+static int ssl_cookie_write(void *ctx, unsigned char **p, unsigned char *end,
```

Comment:
Would it be simpler and more general to require that these functions be implemented in Python, and the user passes them through when wrapping the socket?  Eg:
```py
cookie_dict = {}
def dtls_cookie_handler(ctx, key, value):
    if value is None:
        # create and return new cookie
        cookie = os.urandom(32)
        cookie_dict[key] = cookie
        return cookie
    else:
        # check cookie
        return cookie_dict.get(key) == value
ctx.wrap_socket(..., dtls_cookie=dtls_cookie_handler)
```